### DISCUSSION TITLE
issues with mounted share in Plex LXC - console can see folders but plex app can't

### DISCUSSION THREAD

**Discussion by chenks** [tteck/Proxmox#3045] [Help Needed]
hi, bit of a weird one, but pretty sure i've made a mistake somewhere along the lines.

pve 8.2.2
plex LXC (ubunut unprivileged) created using the current script (https://helper-scripts.com/scripts?id=Plex+Media+Server)

i have my NAS mounted to the PVE at
`/mnt/smbshare/media/` which i can fully access with read/write permissions

i then used
`mp0: /mnt/smbshare/media/,mp=/mnt/media`
within the conf file to mount that inside the LXC

when i connect to the plex lxc console i can browse to /mnt/media and i can see all the folders and read/write no problems.

however when i go into plex and try to add that as a library location plex sees no folders at all
![image](https://github.com/tteck/Proxmox/assets/11945919/e18af744-104a-4eb4-9bae-92011a7f7dd7)

on the console it looks like this
![image](https://github.com/tteck/Proxmox/assets/11945919/8a481300-0b31-474b-a2c0-9ee99bb3b39f)

what have i screwed up?
also, have i made an initial mistake making it unprivileged? i just selected auto when the script ran to created the LXC so i don't remember it asking me, i guess it only would if i used the advanced option.

pve is running on 12th gen intel i3-12100 with 32GB DDR5. i play all my content on current gen appletv's so don't anticipate needing transcoding but ideally would want it enabled regardless (i don't have plex pass though).

**Comment by Mellowlynx** [comment]
Can you click on media?
I think you entered the path in the bar, this will not update the file list on the right.
If you click "media" I think you will see your folders.

Else, click on `/` scroll to and click on `mnt` then click  `media`

Hope this works, everything else is as I have it as well.

**Reply by chenks** [reply]
> Can you click on media?
> 
> I think you entered the path in the bar, this will not update the file list on the right.
> 
> If you click "media" I think you will see your folders.
> 
> 
> 
> Else, click on `/` scroll to and click on `mnt` then click  `media`
> 
> 
> 
> Hope this works, everything else is as I have it as well.

Hi, no I didn't type it, I clicked on mnt then media, and it's empty

**Reply by Mellowlynx** [reply]
Can you remove the share and try to add it as a cifs?
(https://pve.proxmox.com/wiki/Storage:_CIFS)
This is what I use, else it's a challenge to troubleshoot.

First shut down the Plex lxc, then on the PVE host shell run:
`pvesm add cifs <storagename> --server <server> --share <share> [--username <username>] [--password]`

(To see if it works, `less /etc/pve/storage.cfg`, you should see it mounted something like this)
![image](https://github.com/tteck/Proxmox/assets/2348339/b11b15ab-0b38-44ac-af53-3b7eb5f9246d)

Then (also from the PVE shell)
`pct set 103 -mp0 /mnt/pve/media,mp=/mnt/media` This will add the mount point to the LXC.

Now start the Plex LXC and try again to add a new library in Plex.

**Reply by chenks** [reply]
just before i try this..

this is how the nas share is mounted on the pve (via an fstab entry)
`//192.168.50.2/data /mnt/smbshare cifs _netdev,x-systemd.automount,noatime,uid=100000,gid=110000,dir_mode=0770,file_mode=0770,user=******,pass=***** 0 0`

all my other LXCs have no issue accessing it, the only difference i can see with the Plex LXC is that it is Ubuntu whereas all the others are Debian 12 - they all use `mp0: /mnt/smbshare/media/,mp=/mnt/media` in teh conf file the same as the plex conf file

this is why i'm confused as to why it's not working within plex itself (even though it works in the console for the LXC).
is it maybe a permissions issue where the plex app is using a different user compared to root that the console is using?

**Reply by Mellowlynx** [reply]
I did some digging around, and it looks like a permission's problem indeed.
I read somewhere that the LXC must be privileged to use samba.
This can be the reason why mine is working in unprivileged because I added the share different, but I'm not sure about that.

Sadly, I can't test because everything in my full cluster is in production, and I do not have a spare test machine at the moment.
Maybe this tutorial is worth a try: https://forum.proxmox.com/threads/tutorial-unprivileged-lxcs-mount-cifs-shares.101795/
It tries to mount the share as a different user, Plex in your case. 

Else I hope you can find a solution that works for you by looking around online, I found a lot of about the same Plex + NAS problems, so there must be something that can work for you.

**Reply by chenks** [reply]
> I read somewhere that the LXC must be privileged to use samba.

hmm yeah not sure that can be correct though, as all my other LXCs are unprivileged and they can read/write to the mounted smb share just fine.

i suspect the user that the plex app uses is not "root" and that's why the folder appears empty.
however in the deb12 LXCs i get a specific "permission denied" error when there is a permission issues, not sure if ubuntu does it differently and just displays a blank folder instead.

i will dig about some more

also, that link to the tutorial is what i used to originally mount my share.

**Reply by chenks** [reply]
@Mellowlynx it was indeed a permissions issue.
i added the user `plex` to the permissions group and, after a reboot, plex can now correctly see everything in `/mnt/media`

**Reply by Mellowlynx** [reply]
Great, happy you resolved it 😀

**Reply by rucknapucknavitz** [reply]
Someone posted [this script on the proxmox forums](https://forum.proxmox.com/threads/tutorial-unprivileged-lxcs-mount-cifs-shares.101795/page-5#post-668585) that can largely automate this process - but take note of the edit currently needed in Step one to create the lxc_share group.

https://gist.github.com/NorkzYT/14449b247dae9ac81ba4664564669299

**Reply by xaviuchiha** [reply]
I am facing the same situation, how did you add permissions to the user? Thanks!!

**Reply by Kers17** [reply]
> @Mellowlynx it was indeed a permissions issue. i added the user `plex` to the permissions group and, after a reboot, plex can now correctly see everything in `/mnt/media`

I have the same problem. How did you add permissions? Thanks
